rewrite/to_ir.rs

```diff
@@ -119,109 +41,145 @@ impl<'i> ParseTreeVisitor<'i, CypherParserContextType> for FilterVisitor {
 }
 
 impl<'i> CypherVisitor<'i> for FilterVisitor {
-    /// This function visits a match clause.  It extracts the graph inside, complete with any
-    /// node attributes, and stores that information in a struct_filter.  It then extracts any information
-    /// in the where clause, which pertains to the whole graph, and stores that in an attribute_filter.
-    fn visit_oC_Match(&mut self, ctx: &OC_MatchContext<'i>) {
+    // we do not want to visit returns in this case, ignore that part of the tree
+    // TODO: Apply the visitor directly to the MATCH body instead
+    fn visit_oC_Return(&mut self, _ctx: &OC_ReturnContext<'i>) {}
+
+    fn visit_oC_PropertyOrLabelsExpression(
+        &mut self,
+        ctx: &OC_PropertyOrLabelsExpressionContext<'i>,
+    ) {
+        log::info!("Generating IrReturn");
+        let atom = ctx.oC_Atom().unwrap();
+        let entity: String;
+        if let Some(func) = atom.oC_FunctionInvocation() {
+            entity = func.get_text();
+            // TODO: We can make this UDF more precise
+            log::info!("Storing UDF: {:?}", entity);
+        } else if let Some(var) = atom.oC_Variable() {
+            entity = var.get_text();
+            log::info!("Storing var: {:?}", entity);
+        } else if let Some(var) = atom.oC_Literal() {
+            entity = var.get_text();
+            log::info!("Storing literal: {:?}", entity);
+        } else {
+            log::error!(
+                "Unsupported expression {:?}. Has type {:?}",
+                atom.get_text(),
+                ruleNames[atom.get_child(0).unwrap().get_rule_index()]
+            );
+            process::exit(1);
+        }
+
+        let mut property_str = String::new();
+        for property in ctx.oC_PropertyLookup_all() {
+            // this includes the dots
+            property_str.push_str(&property.get_text())
```

Comment:
I fixed this in my branch but that's nowhere near ready to push yet - can we switch from property.get_text() to property.oC_PropertyKeyName().unwrap().get_text()

That gets rid of the dots, and the dots are rather annoying once you try to generate code.

Reply:
Can do, but something like "node.metadata.name" will end up like "metadataname" then.

Reply:
Okay, then let's leave it with preceding dots for now and we can figure out where to clean it up later.